src/main/java/com/internet/shop/controllers/DeleteProductController.java

```diff
@@ -0,0 +1,23 @@
+package com.internet.shop.controllers;
+
+import com.internet.shop.lib.Injector;
+import com.internet.shop.service.ProductService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class DeleteProductController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final ProductService productService =
+            (ProductService) injector.getInstance(ProductService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Long productId = Long.valueOf(req.getParameter("id"));
+        productService.delete(productId);
+        resp.sendRedirect(req.getContextPath() + "/product/all/admin");
```

Comment:
let's do the same after we add a new product to storage in AddProductController: `resp.sendRedirect(req.getContextPath() + "/product/all/admin")`

Reply:
done